tests/components/lock/test_init.py

```diff
@@ -0,0 +1,12 @@
+"""The tests for Lock."""
+from homeassistant.components import lock
+
+
+def test_deprecated_base_class(caplog):
+    """Test deprecated base class."""
+
+    class CustomLock(lock.LockDevice):
+        pass
+
+    CustomLock()
+    assert "LockDevice is deprecated, modify CustomLock" in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5418277)